### PR TITLE
member: delete its dc-location info while deleting a member

### LINKED ISSUE #3239: Update the pd-ctl to support the dc-location related operations
### Describe your feature request related problem

For now, to manage the `dc-location` of a PD cluster, we can only modify the configuration. And also we don't have an easy way to delete or modify the old `dc-location` from the etcd for the users.

### Describe the feature you'd like

Update the `pd-ctl` by adding some commands to support the `dc-location` related operations.

### PR DESCRIPTION
**Pull request by Riptide3**
Signed-off-by: Riptide3 <riptide.yzy@gmail.com>

### What problem does this PR solve?

Close #3239. In this issue we want a method (e.g. a `pd-ctl` command) to delete the old `dc-location` info persisted in the etcd, however, we may only use it when a PD is deleted.

This PR modified the process of deleting a PD member to automatically delete its `dc-location` info also. So we can achieve the same goal without introducing a redundant command.

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: server/api/member.go [hunk 1]
```diff
@@ -133,6 +133,13 @@ func (h *memberHandler) DeleteByName(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Delete dc-location.
+	err = h.svr.GetMember().DeleteDCLocation(id)
```
Comment (JmPotato) [inline server/api/member.go:137]: Maybe `DeleteMemberDCLocationInfo` is a better name?

File: server/api/member.go [hunk 2]
```diff
@@ -165,6 +172,13 @@ func (h *memberHandler) DeleteByID(w http.ResponseWriter, r *http.Request) {
...
```
Comment (JmPotato) [inline server/api/member.go:176]: Ditto.

File: server/member/member.go [hunk 3]
```diff
@@ -297,6 +298,14 @@ func (m *Member) getMemberLeaderPriorityPath(id uint64) string {
 	return path.Join(m.rootPath, fmt.Sprintf("member/%d/leader_priority", id))
 }
 
+func (m *Member) GetDCLocationPathPrefix() string {
+	return path.Join(m.rootPath, dcLocationConfigEtcdPrefix)
+}
+
+func (m *Member) GetDCLocationPath(id uint64) string {
+	return path.Join(m.rootPath, dcLocationConfigEtcdPrefix, fmt.Sprint(id))
```
Comment (JmPotato) [inline server/member/member.go:306]:
```suggestion
	return path.Join(m.GetDCLocationPathPrefix(), fmt.Sprint(id))
```

I think this is better.

File: server/member/member.go [hunk 4]
```diff
@@ -297,6 +298,14 @@ func (m *Member) getMemberLeaderPriorityPath(id uint64) string {
 	return path.Join(m.rootPath, fmt.Sprintf("member/%d/leader_priority", id))
 }
 
+func (m *Member) GetDCLocationPathPrefix() string {
```
Comment (JmPotato) [inline server/member/member.go:301]: Please add some comments for this method.

File: server/member/member.go [hunk 5]
```diff
@@ -323,6 +332,19 @@ func (m *Member) DeleteMemberLeaderPriority(id uint64) error {
 	return nil
 }
 
+// DeleteDCLocation removes a member's dc-location.
+func (m *Member) DeleteDCLocation(id uint64) error {
```
Comment (JmPotato) [inline server/member/member.go:336]: Same as the first suggestion.

**Review by JmPotato — COMMENTED** [review]
Thanks for your contribution! Please address these review comments.

**Comment by Riptide3** [comment]
> Thanks for your contribution! Please address these review comments.

I have addressed these comments. PTAL

**Review by JmPotato — APPROVED** [review]
LGTM

**Review by Yisaer — COMMENTED** [review]
What about adding a unit test?

**Comment by Riptide3** [comment]
> What about adding a unit test?



> What about adding a unit test?

Ok. The test is already added. 😃

**Review by rleungx — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]
LGTM

**Pull request merged**